src/Microsoft.OData.Core/Json/ODataUtf8JsonWriter.Stream.cs

```diff
@@ -0,0 +1,295 @@
+            /// </summary>
+            /// <param name="writer">The OData UTF-8 JSON writer to write to.</param>
+            internal ODataUtf8JsonWriterStream(ODataUtf8JsonWriter writer)
+            {
+                this.jsonWriter = writer;
+            }
+            public override bool CanRead => false;
+
+            public override bool CanSeek => false;
+
+            public override bool CanWrite => true;
+
+            public override long Length => throw new NotImplementedException();
+
+            public override long Position { get => throw new NotImplementedException(); set => throw new NotImplementedException(); }
+
+            /// <summary>
+            /// Flushes any buffered data to the underlying stream synchronously.
+            /// </summary>
+            public override void Flush()
+            {
+                // If there are unprocessed bytes, encode and write them as the final block.
+                ReadOnlySpan<byte> bytesNotProcessedFromPreviousChunk = buffer;
+
+                if (!bytesNotProcessedFromPreviousChunk.IsEmpty)
+                {
+                    this.jsonWriter.Base64EncodeAndWriteChunk(bytesNotProcessedFromPreviousChunk, isFinalBlock: true, out int bytesNotWrittenFromPreviousChunk);
+                    // Clear the buffer since all bytes have been written.
+                    buffer = Array.Empty<byte>();
+                }
+
+                this.jsonWriter.Flush();
+            }
+
+            /// <summary>
+            /// Flushes any buffered data to the underlying stream asynchronously.
+            /// </summary>
+            /// <param name="cancellationToken">A CancellationToken to observe while waiting for the flush operation to complete.</param>
+            /// <returns>A task representing the asynchronous flush operation.</returns>
+            public override async Task FlushAsync(CancellationToken cancellationToken)
+            {
+                // If there are unprocessed bytes, encode and write them as the final block.
+                ReadOnlyMemory<byte> bytesNotProcessedFromPreviousChunk = buffer;
+
+                if (!bytesNotProcessedFromPreviousChunk.IsEmpty)
+                {
+                    this.jsonWriter.Base64EncodeAndWriteChunk(bytesNotProcessedFromPreviousChunk.Span, isFinalBlock: true, out int bytesNotWrittenFromPreviousChunk);
+                    // Clear the buffer since all bytes have been written.
+                    buffer = Array.Empty<byte>();
+                }
+
+                await this.jsonWriter.FlushAsync();
+            }
+
+            public override int Read(byte[] buffer, int offset, int count)
+            {
+                throw new NotImplementedException();
+            }
+
+            public override long Seek(long offset, SeekOrigin origin)
+            {
+                throw new NotImplementedException();
+            }
+
+            public override void SetLength(long value)

```

Comment:
To provide the user of the streaming API with helpful tips:
```suggestion
            /// <summary>
            /// Sets the length of the stream. This operation is not supported by this stream.
            /// </summary>
            public override void SetLength(long value)
```